Chigusa Hydroelectric Dam - A damn big port.

RipGrayson opened the pull request:
<!-- ***STOP!***  Read this: If this is not a PR ready for review and merge or WIP, open it as a draft PR, using the arrow next to 'Create Pull Request'>

<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

Ports the Trijent map from CM into TGMC with very minor map alterations. Provided this PR is merged, I'll use it as a base and slowly alter it to be more TGMC friendly.
I'm going to post the answers to a few questions here just to save time.

**Q:** Sounds good but how many CM assets does this use?
**A:** Some asphalt textures and one road decal. Everything else is stock TGMC or made by me.
**Q:** Can we do this? 
**A:** CM licenses their art under Creative Commons 3.0 BY-SA license, which allows for free use of art assets. If it somehow raises a fuss it's simple to redraw the small amount of CM textures this PR uses. In the event the map itself receives an official complaint, or a maintainer is uncomfortable I'll redo the entire thing to be more of a homage. (Which is probably what it will be post rework anyway.)
**Q:** Will I die to polluted water crossing a river?
**A:** Toxic water is not a feature that I ported, water filtration is nonfunctional and only cosmetic. 
**Q:** Will I see this map in rotation? Trijent _is_ pretty damn large...
**A.** Probably not frequently at first, the long term plan is to reduce it in size until it can comfortably fit 70-120 players. Even post-merge consider this map under active development.

## Why It's Good For The Game

More maps = more variety, provides a high-pop map for those times we go over 100 players. Gives us a kickass place to ride motorbikes and have HvH events. 

## Changelog
:cl:
add: Added Shigusa Hydroelectric Dam, a map for more than 100 players.
imageadd: Added a new couple icons for shallow water transitions, broken generators, and stairs.
code: Cleaned up some unused turfs.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

vilereaver41 commented:
this is a disaster waiting to happen during the first few days

BraveMole reviewed (commented):
So some remarks, i haven't reviewed anything yet, i'll probably do it tonight :

- Have you checked https://github.com/tgstation/TerraGov-Marine-Corps/pull/8104 ? Check that you are following these, we are just formalizing what was already enforced (more or less). If you have question about ai nodes, feel free to ping me on discord
- It's a pity that your map is only for over 90 players, because we don't have this pop. Consider shortening the map and making it medium-high pop (above 50-70), or it will never be played and it's a shame. Ultimatly, it's your map, so you do whatever you want
- Give credit to the author of the map, at least the orginal one (link the cm pr if you find, if you don't find the pr, just try to find the name of the author and ping him, it's always nice to get some credit)

Overall, very very nice to see a new map, good work

RipGrayson commented:
> So some remarks, i haven't reviewed anything yet, i'll probably do it tonight :
> 
> * Have you checked [Map contrib guidelines added #8104](https://github.com/tgstation/TerraGov-Marine-Corps/pull/8104) ? Check that you are following these, we are just formalizing what was already enforced (more or less). If you have question about ai nodes, feel free to ping me on discord
I'll look into AI nodes after I fix the compile map errors.
> * It's a pity that your map is only for over 90 players, because we don't have this pop. Consider shortening the map and making it medium-high pop (above 50-70), or it will never be played and it's a shame. Ultimatly, it's your map, so you do whatever you want
This is something of a base, once it's in I'll work towards making the map more friendly towards our level of population.
> * Give credit to the author of the map, at least the orginal one (link the cm pr if you find, if you don't find the pr, just try to find the name of the author and ping him, it's always nice to get some credit)
Trijent predates TGMC by a bit (we actually have early versions of some of the prerelease assets still floating around) and was made by a team of CM mappers while CM was still closed source. I was unable to find any information on who worked on it, and there's no gitlab PR to credit the original submitter.

BraveMole commented:
Undraft when ready for review

RipGrayson commented:
This should be ready for review.

BraveMole reviewed (changes requested):
Overall, good work, i'm really exciting about playing on this map!

![image](https://user-images.githubusercontent.com/25491240/131408806-fe09acec-56c2-4fb3-bc48-58d7a4d15e29.png)
 I'm afraid this crash site is out of the map? 
 
 
![image](https://user-images.githubusercontent.com/25491240/131408902-2bf14c8a-b33f-418a-8233-2a40efa09731.png)

This is symptomatic of your nodes placement. It's not how you should do it, this will confuse ai the hell out. Please contact me on discord on how to place those.

![image](https://user-images.githubusercontent.com/25491240/131409385-6865777f-dcf5-4ba0-ae16-ea8baaa69013.png)

What is this?

The LV needs to be marked (lv1, lv2) and also need to have shuttle computer. I suggest you use the landmarks one, so i won't have to change them if i make this map hvh compatible.

![image](https://user-images.githubusercontent.com/25491240/131409947-c30b7122-f0be-4052-8026-b0c470e8e169.png)

in all this area, xeno will be unable to build before shuttle. It's too big. Take into account the anti-wall spam protection.

![image](https://user-images.githubusercontent.com/25491240/131410039-9a59a0d5-9d3d-43f2-828d-f805edf40462.png)

Same remark, the areas need improvement here. I don't see any apc on the lzs? Fob should have their own area, there should be small areas around it, and then the rest is fine

![image](https://user-images.githubusercontent.com/25491240/131410302-bc3915ae-24ed-4b0e-aece-7c04ea7d4a9e.png)

Ugly as hell, use our catwalks

![image](https://user-images.githubusercontent.com/25491240/131410359-2bd43f32-76fd-4e7b-b42a-7f0a6c302235.png)

Weird patch of marking

![image](https://user-images.githubusercontent.com/25491240/131410454-3e54abdd-b905-45b1-82f8-9f84f65ccb31.png)
 
this part feels really empty compared to the rest, can you fill it with stuff? Or just delete that part, the map is already huge.

![image](https://user-images.githubusercontent.com/25491240/131410586-2fef558b-a22f-4adc-8f27-0557105777e0.png)

This is too much of a dead end for me, add an exit on the south/north or just delete

![image](https://user-images.githubusercontent.com/25491240/131410774-e49819d1-3f3e-4052-a0ce-652b674ae943.png)

Try to lower the amount of gens on the map to around 10-15. Also, it might be interesting to have two small engie part rather than one, it gives more objectives.

I see no intelligence computer, i know no one is make those missions but still, add a bunch.

RipGrayson commented:
> Overall, good work, i'm really excited about playing on this map!
> 
> ![image](https://user-images.githubusercontent.com/25491240/131408806-fe09acec-56c2-4fb3-bc48-58d7a4d15e29.png)
> I'm afraid this crash site is out of the map?
> 
Fixed
> ![image](https://user-images.githubusercontent.com/25491240/131408902-2bf14c8a-b33f-418a-8233-2a40efa09731.png)
> 
> This is symptomatic of your nodes placement. It's not how you should do it, this will confuse ai the hell out. Please contact me on discord on how to place those.
> 
Yeah I'll do that in a bit.
> ![image](https://user-images.githubusercontent.com/25491240/131409385-6865777f-dcf5-4ba0-ae16-ea8baaa69013.png)
> 
> What is this?
> 
I think CM has a mechanic where you can pilot escape pods off the planet. They stick the pods in there during transit for some reason, probably a lack of TG shuttle system. I removed the rooms, they have no use here.
> The LV needs to be marked (lv1, lv2) and also need to have shuttle computer. I suggest you use the landmarks one, so i won't have to change them if i make this map hvh compatible.
> 
> ![image](https://user-images.githubusercontent.com/25491240/131409947-c30b7122-f0be-4052-8026-b0c470e8e169.png)
> 
In their infinite wisdom CM placed the shuttle computers north of LZ1 (the north one) in a small building, and just east of LZ2 in a small office directly adjacent. I didn't bother changing the positions but I did make sure both are accessible and protected by shutters. I can easily move both of them, I was just pointing out that both LZs technically have computers.
> in all this area, xeno will be unable to build before shuttle. It's too big. Take into account the anti-wall spam protection.
> 
Reduced the size by about half.
> ![image](https://user-images.githubusercontent.com/25491240/131410039-9a59a0d5-9d3d-43f2-828d-f805edf40462.png)
> 
> Same remark, the areas need improvement here. I don't see any apc on the lzs? Fob should have their own area, there should be small areas around it, and then the rest is fine
> 
Come to think of it, apparently CM has no APCs on their LZs. Weird design decision. I added one to both and expanded the external FOB areas.
> ![image](https://user-images.githubusercontent.com/25491240/131410302-bc3915ae-24ed-4b0e-aece-7c04ea7d4a9e.png)
> 
> Ugly as hell, use our catwalks
> 
Done.
> ![image](https://user-images.githubusercontent.com/25491240/131410359-2bd43f32-76fd-4e7b-b42a-7f0a6c302235.png)
> 
> Weird patch of marking
> 
Fixed.
> ![image](https://user-images.githubusercontent.com/25491240/131410454-3e54abdd-b905-45b1-82f8-9f84f65ccb31.png)
> 
> this part feels really empty compared to the rest, can you fill it with stuff? Or just delete that part, the map is already huge.
> 
This area used to be filled with filtration equipment, since my philosophy was to avoid using as much CM art as possible I just blanked the areas where spritework was. I'll either replace the area, or delete it entirely soon.
> ![image](https://user-images.githubusercontent.com/25491240/131410586-2fef558b-a22f-4adc-8f27-0557105777e0.png)
> 
> This is too much of a dead end for me, add an exit on the south/north or just delete
> 
Done.
> ![image](https://user-images.githubusercontent.com/25491240/131410774-e49819d1-3f3e-4052-a0ce-652b674ae943.png)
> 
> Try to lower the amount of gens on the map to around 10-15. Also, it might be interesting to have two small engie part rather than one, it gives more objectives.
> 
Believe it or not during testing it takes that many gens to fully power Trijent. I'll reduce it down to 12 and see how it does.
> I see no intelligence computer, i know no one is make those missions but still, add a bunch.
>
Added 25 intel computers in various areas of the map.

AndroidMK2 commented:
Seems very epic would definetly want to play on it 😳

bowlofcereal commented:
This is actually fucking cool dude.

BraveMole commented on code/__DEFINES/minimap.dm:
```diff
@@ -27,10 +27,15 @@ GLOBAL_LIST_INIT(all_minimap_flags, bitfield2list(MINIMAP_FLAG_ALL))
+#define MINIMAP_AREA_RIVER "#25348bee"
+#define MINIMAP_AREA_SCIENCE "#8d13bdee"
+#define MINIMAP_AREA_BOTANY "#2bf119ee"
+#define MINIMAP_AREA_CIVILIAN "#633705ee"
```
you are not using this. This is out of scope for the PR anyway, please remove

BraveMole reviewed (changes requested):
Solid work, appart from these two remarks and the nodes it's fine for me.

![image](https://user-images.githubusercontent.com/25491240/131568250-839898f6-8be7-40c3-9e0c-e6d175565781.png)

You need shutters all around the lz, because marines are dumb and will PC the rock to get out

![image](https://user-images.githubusercontent.com/25491240/131568638-0a4c74fd-2458-458f-8bc7-3c4b536c3060.png)

Weird empty space

AnCopper commented:
Trijent without the poop water is like bread without butter, smh..

RipGrayson commented:
> Trijent without the poop water is like bread without butter, smh..

If people want the toxic water mechanics back I'll look into restoring it post merge. I'm not currently sensing a great deal of demand for it though.

mrhugo13 commented:
Need to add miners, without it things economy side get a bit ehhhhhhhhhh.

BraveMole commented on _maps/map_files/desertdam/desertdam.dmm:
```diff
@@ -18945,6 +18951,7 @@
 	dir = 1
 	},
 /obj/structure/platform{
+	density = 0;
```
It's not allowed to make vv on map objects

BraveMole reviewed (changes requested):
So i see you added miners, good.

Unfortunatly, the map is way too big for our map, so i propose two options: 

- We freeze this map until we get vehicles, which could take from 1 to 3 months (apc, things like that). And then we give a ton of free vehicles, and a lot of already spawned tunnels for xeno
- You cut 30-50% of the map

RipGrayson commented:
> So i see you added miners, good.
> 
> Unfortunatly, the map is way too big for our map, so i propose two options:
> 
> * We freeze this map until we get vehicles, which could take from 1 to 3 months (apc, things like that). And then we give a ton of free vehicles, and a lot of already spawned tunnels for xeno
> * You cut 30-50% of the map

I'm no illusion that this map is perfect, it's a map made for a different codebase with minimal changes.
Unfortunately I cannot cut 50% of the map quickly, it would take me at least a month to do it. This map is a base, I was planning on incrementally implementing size and balance changes after merge, collecting feedback on each change.

I guess if you want we could just add the map and let it be inaccessible to all but admins while I work on improving it.

BraveMole commented:
> I'm no illusion that this map is perfect, it's a map made for a different codebase with minimal changes.
> Unfortunately I cannot cut 50% of the map quickly, it would take me at least a month to do it. This map is a base, I was planning on incrementally implementing size and balance changes after merge, collecting feedback on each change.
> 
> I guess if you want we could just add the map and let it be inaccessible to all but admins while I work on improving it.

No rush, since it's not really votable due to the pop config, i can merge without issue. Honestly i think the vehicle idea might make this map really unique, and tivi is on that

BraveMole reviewed (changes requested):
I'll merge when the plateform problem is fixed.

Code wise, it shouldn't be that hard, you could do this in another pr. Otherwise, you can remove all plateform but that would be a shame;

Another note, this miner is too close to the lz
![image](https://user-images.githubusercontent.com/25491240/131974819-6c10e7ef-06bd-4452-b5c6-2778b9866560.png)
Remove it please

RipGrayson commented:
Well platforms have been fixed, I don't think there's anything else in the way of preventing a merge.
